File: Problems/Algorithms/406. Queue Reconstruction by Height/QueueReconstruction.java
class Solution {
    public int[][] reconstructQueue(int[][] people) {
        Arrays.sort(people, (a, b) -> a[0] != b[0] ? b[0] - a[0] : a[1] - b[1]);
        
        List<int[]> results = new LinkedList<>();
        for (int[] curr : people) {
            results.add(curr[1], curr);
        }
        
        return results.toArray(new int[people.length][]);
    }
}

File: Problems/Algorithms/406. Queue Reconstruction by Height/queue_reconstruction.rs
impl Solution {
    pub fn reconstruct_queue(mut people: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        people.sort_by_key(|p| (-p[0], p[1]));
        
        let mut results: Vec<Vec<i32>> = Vec::new();
        for p in people.iter() {
            results.insert(p[1] as usize, p.to_vec());
        }
        results
    }
}

